### DISCUSSION TITLE
external database settings are not compatible with PlanetScale: "gtid_mode" must be "ON", but found: "OFF"

### DISCUSSION THREAD

**Discussion by ghost** [planetscale/discussion#522] [Q&A] [answered]
I want to do a Database Import from MySQL built on VPS to PlanetScale.
But when I try to do that, I get the following error:
```
external database settings are not compatible with PlanetScale: "gtid_mode" must be "ON", but found: "OFF"
```

mysql  Ver 8.0.34-0ubuntu0.22.04.1 for Linux on x86_64 ((Ubuntu))

MySQL is using version 8, and the settings remain pristine.
I think the binary log is enabled because binlog exists in the "/var/lib/mysql" path.
But you probably don't have a transaction ID in that log.
So I looked for a way to enable gtid_mode, but couldn't find one

What I tried:
I wrote the following on "/etc/mysql/my.cnf"
```
[mysqld]
gtid_mode = ON
```

**Comment by mcrauwel** [comment, marked as answer]
Hey @KohnoseLami 

Check the MySQL manual for the full explanation: https://dev.mysql.com/doc/refman/8.0/en/replication-mode-change-online-enable-gtids.html

TL;DR: 
- you need `gtid_mode = ON` and `enforce_gtid_consistency = ON`
- changing the my.cnf will require you to restart the MySQL instance
- if you want to enable GTID in an online way you need to go from `OFF` -> `OFF_PERMISSIVE` -> `ON_PERMISSIVE` -> `ON`

**Reply by ghost** [reply]
![image](https://github.com/planetscale/discussion/assets/64224091/727a3199-48ae-46a3-ae6c-df80e7dc693a)
Thank you!
It's solved.

It worked by enabling two options
